phoenix-core/src/it/java/org/apache/phoenix/end2end/BaseJoinIT.java

```diff
@@ -30,14 +30,19 @@
 import java.util.Map;
 import java.util.regex.Pattern;
 
+import org.apache.hadoop.hbase.HConstants;
+import org.apache.phoenix.cache.ServerCacheClient;
+import org.apache.phoenix.end2end.HashJoinCacheIT.InvalidateHashCacheRandomly;
+import org.apache.phoenix.util.ReadOnlyProps;
 import org.apache.phoenix.util.SchemaUtil;
 import org.apache.phoenix.util.StringUtil;
 import org.junit.Before;
+import org.junit.BeforeClass;
 
 import com.google.common.collect.ImmutableMap;
 import com.google.common.collect.Maps;
 
-public abstract class BaseJoinIT extends ParallelStatsDisabledIT {
+public abstract class BaseJoinIT extends BaseUniqueNamesOwnClusterIT {
```

Comment:
Can we keep BaseJoinIT derived from ParallelStatsDisabledIT? Otherwise, there's a lot more mini clusters created which increases our end-to-end test time run more and more.

If the reason is because you're adding InvalidateHashCacheRandomly, just add it instead dynamically to the table created in your test. To prevent flapping, you need to poll until you the descriptor matches what you've changed. See PartialIndexRebuilderIT.addWriteFailingCoprocessor() for an example you can generalize and/or copy/paste.